compiler/core/js_dump.ml

```diff
@@ -524,6 +524,48 @@ and expression_desc cxt ~(level : int) f x : cxt =
          when Ext_list.length_equal el i
        ]}
     *)
+  | Call
+      ( ({expression_desc = J.Var (J.Qualified (_, Some fnName))} as e),
+        el,
+        {call_transformed_jsx = true} ) -> (
+    match el with
+    | [
+     tag;
+     {
+       expression_desc =
+         Caml_block (el, _mutable_flag, _, Lambda.Blk_record {fields});
+     };
+    ] ->
+      let fields =
+        Ext_list.array_list_filter_map fields el (fun (f, opt) x ->
+            match x.expression_desc with
+            | Undefined _ when opt -> None
+            | _ -> Some (f, x))
+      in
+      print_jsx cxt ~level f fnName tag fields
+    | [
+     tag;
+     {
+       expression_desc =
+         Caml_block (el, _mutable_flag, _, Lambda.Blk_record {fields});
+     };
+     key;
+    ] ->
+      let fields =
+        Ext_list.array_list_filter_map fields el (fun (f, opt) x ->
+            match x.expression_desc with
+            | Undefined _ when opt -> None
+            | _ -> Some (f, x))
+      in
+      let fields = ("key", key) :: fields in
+      print_jsx cxt ~level f fnName tag fields
+    | _ ->
+      expression_desc cxt ~level f
```

Comment:
Is this code branch ever executed?

Reply:
If it is, it means we missed something. Could we add some sort of assert in this case?